pkg/prometheus/promcfg.go

```diff
@@ -1231,9 +1240,15 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 
 	// Filter targets based on correct port for the endpoint.
 	if ep.Port != "" {
+		var sourceLabelItem yaml.MapItem
+		if endpointSliceSupported {
+			sourceLabelItem = yaml.MapItem{Key: "source_labels", Value: []string{"__meta_kubernetes_endpointslice_port_name"}}
+		} else {
+			sourceLabelItem = yaml.MapItem{Key: "source_labels", Value: []string{"__meta_kubernetes_endpoint_port_name"}}
+		}
```

Comment:
(nit) a bit less duplication IMO
```suggestion
		sourceLabels := []string{"__meta_kubernetes_endpoint_port_name"}
		if endpointSliceSupported {
		        sourceLabels = []string{"__meta_kubernetes_endpointslice_port_name"}
		}
```